iceoryx_hoofs/source/cxx/requires.cpp

```diff
@@ -19,15 +19,17 @@
 #include "iceoryx_hoofs/error_handling/error_handling.hpp"
 
 #include <iostream>
+#include <cstdint>
 
 namespace iox
 {
 namespace cxx
 {
 namespace internal
 {
+// AXIVION Next Line  AutosarC++19_03-A3.3.1 : The function is declared with in the namespace
 void Require(
-    const bool condition, const char* file, const int line, const char* function, const char* conditionString) noexcept
+    const bool condition, const int8_t* file, const int32_t line, const int8_t* function, const int8_t* conditionString) noexcept
```

Comment:
Not quite sure about this change. `file`, `line` and `function` will be fed with the data from the builtin compiler macros `__FILE__`, `__LINE__` and `__FUNCTION__` which are of non fixed width type.

Similar with `conditionString`. Our codebase uses `const char` for all string literals and fixed width types only for integer. Do we have to change this?

Reply:
This does not compile, see:
```cxx
void f(const int8_t* someString) {
  std::cout << someString << std::endl; 
}

f("hello world"); // this will be deduced as `const char*` so the compiler will state `no matching function`
f(__FILE__); // will also be deduced as `const char*` (this is how we use this)
```

My suggestion would be to use `const char*` instead of `const int8_t*` since they are two different types.

Reply:
@elfenpiff exactly. The typedefs are `uint8_t -> unsigned char` and `int8_t -> signed char` with `char != signed char`

Reply:
@elfenpiff agreed `char != signed char` shall consider `const char*` only here.